core/corehttp/metrics_test.go

```diff
@@ -44,11 +44,13 @@ func TestPeersTotal(t *testing.T) {
 
 	node := &core.IpfsNode{PeerHost: hosts[0]}
 	collector := IpfsNodeCollector{Node: node}
-	actual := collector.PeersTotalValues()
-	if len(actual) != 1 {
-		t.Fatalf("expected 1 peers transport, got %d, transport map %v", len(actual), actual)
+	peersTransport := collector.PeersTotalValues()
+	if len(peersTransport) > 2 {
+		t.Fatalf("expected at most 2 peers transport (tcp and upd/quic), got %d, transport map %v",
+			len(peersTransport), peersTransport)
 	}
-	if actual["/ip4/tcp"] != float64(3) {
-		t.Fatalf("expected 3 peers, got %f", actual["/ip4/tcp"])
+	totalPeers := peersTransport["/ip4/tcp"] + peersTransport["/ip4/udp/quic"]
+	if totalPeers != 3 {
```

Comment:
I think this test is fine. However, I'll just note that it still depends on internals of libp2p. This test is creating "contact" to 3 peers, but it does not control what type of connections (and how many) are actually created to each peer (this is libp2p's business). Yet, this test checks for specific types of connections. So, if libp2p were to change and e.g. drop the quic transport, this test will fail again. In any event, I think the test is fine as it is, since it is unlikely libp2p will drop tcp or quic anytime soon.

Reply:
Agreed. I tried to retain the spirit of the test since this metric provides a map that collects transports, not just peer counts, but I'm also fine removing this level of specificity.